sky/clouds/aws.py

```diff
@@ -314,9 +316,20 @@ def check_credentials(self) -> Tuple[bool, Optional[str]]:
             elif line.startswith('secret_key'):
                 if '<not set>' not in line:
                     secret_key_ok = True
+
         if access_key_ok and secret_key_ok:
+            # Checks if AWS credentials are valid and can connect to AWS services.
+            # https://stackoverflow.com/questions/53548737/verify-aws-credentials-with-boto3
+            sts = boto3.client('sts')
+            try:
+                sts.get_caller_identity()
+            except boto3.exceptions.ClientError:
+                return False, (
+                    'AWS credentials are not set properly.'
+                    ' Make sure that the access and secret keys are correct'
+                    ' in `~/.aws/credentials`.')
```

Comment:
With these lines, do we still need L309-L318?

Also, for the logs, please add the instruction for configure the credentials. Probably, `'AWS credentials ...'+ help_str`.

Reply:
Regarding line 309-318, it just checks if the credentials are "well-formed". Even if the credentials are not valid, it will show: 
```
      Name                    Value             Type    Location
      ----                    -----             ----    --------
   profile                <not set>             None    None
access_key     ****************asdf shared-credentials-file    
secret_key     ****************asdf shared-credentials-file    
    region                us-east-2      config-file    ~/.aws/config
```
Hence, there are two layers of check, 
1) Check if credentials are well formed (309-318)
2) Check if credentials are valid (this PR)

For second comment, done thanks!

Reply:
Just curious, doesn't the second layer also check the "well-formed" for the credential file? Since we do not distinguish much of the hints for the two layers of checks, can we only use the second one?

Reply:
I think it is fine for now. Double layer of security. It is possible that the user runs `aws configure` but doesn't input anything. Also it is a great abstraction to separate both.

Reply:
Sounds good, though I still think only using the `boto3` is enough by adding another `except NoCredentialsError` block becuase the first layer check is very manual and could cause problem for any possible format changes in the future version of aws-cli. As long as the credential file pass the `boto3` check, it is valid for launching the cluster, right?

Also, how about changing the current hints to:
```
  'Failed to access AWS with credentials in ~/.aws/credentials.'
  ' Make sure that the access and secret keys are correct.'
  ' To reconfigure the credentials, ' + help_str[0].lower() + help_str[1:])
```

Reply:
Done, thanks!